app/pages/publications-page.cjsx

```diff
@@ -0,0 +1,331 @@
+    currentSort: 'showAll'
+
+  componentDidMount: ->
+    button = React.findDOMNode(@refs.showAll)
+    @updateButtonState button
+
+  render: ->
+    sideBarNav = counterpart "publications.nav"
+    <div className="publications-page secondary-page-copy">
+      <aside className="secondary-page-side-bar">
+        <nav ref="sideBarNav">
+          {for navItem of sideBarNav
+            <button key={navItem} ref={navItem} className="secret-button side-bar-button" onClick={@showPublicationsList.bind(null, navItem)}><Translate content="publications.nav.#{navItem}" /></button>}
+        </nav>
+      </aside>
+      <section className="publications-content">
+        <h2>{if @state.currentSort is 'showAll'
+              <Translate content="publications.content.header.showAll" />
+            else
+              @state.currentSort
+        }</h2>
+        {for category, projects of publicationCategories
+          if (@state.currentSort is category) or (@state.currentSort is 'showAll')
+            <ul key={category} className="publications-list">
+              {for project in projects
+                <div key={project.name || project.slug}>
+                  <PromiseRenderer promise={apiClient.type('projects').get(slug: project.slug)} pending={null} catch={null}>{([fetchedProject]) =>
+                    if fetchedProject?
+                      <h3 className="project-name">{fetchedProject.display_name}</h3>
+                  }</PromiseRenderer>
+                  {if project.name? then <h3 className="project-name">{project.name}</h3>}<span className="publication-count">{' '}({project.publications.length})</span>
+                  {projectAvatar = @getAvatar(project)
+                  project.publications.map (publication) =>
+                    i = Math.random()
+                    <li key="publication-#{i}" className="publication-item">
+                      <img src={if projectAvatar? then projectAvatar else "./assets/simple-pattern.jpg"} alt="Project avatar" />
+                      <div className="citation">
+                        <p>
+                          <cite>{publication.citation}</cite><br />
+                          {if publication.href? then <a href={publication.href} target="_blank">Available here.</a>}
+                        </p>
+                      </div>
+                    </li>}
+                </div>
+            }</ul>
+        }
+      </section>
+    </div>
+
+  showPublicationsList: (navItem) ->
+    currentButton = React.findDOMNode(@refs[navItem])
+    @setState currentSort: navItem, -> @updateButtonState(currentButton)
+
+  updateButtonState: (currentButton) ->
+    buttons = React.findDOMNode(@refs.sideBarNav).childNodes
+    for button in buttons
+      button.classList.remove 'active'
+    currentButton.classList.add 'active'
+
+  getAvatar: (project) ->
+    apiClient.type('projects').get(slug: project.slug).get('avatar')
+      .then (avatar) => @avatarSrc = avatar.src
+    @avatarSrc
```

Comment:
This isn't going to return the expected value. Might need a PromiseRenderer in the render method.